Sources/PointFree/Account/Team.swift

```diff
@@ -9,24 +9,36 @@ import Prelude
 import Styleguide
 import Tuple
 
-let leaveTeamMiddleware
+let leaveTeamMiddleware: M<Tuple2<User?, SubscriberState>>
   = requireOwner
     <<< leaveTeam
     <| redirect(
       to: .account(.index),
       headersMiddleware: flash(.notice, "You are no longer a part of that team.")
 )
 
-private let requireOwner
-  : MT<Tuple2<User?, SubscriberState>, Tuple2<User, SubscriberState>>
-  = filterMap(require1 >>> pure, or: loginAndRedirect)
+let joinTeamLandingMiddleware: M<Tuple3<User?, SubscriberState, User.TeamInviteCode>>
+  = requireOwner
+    <| writeStatus(.ok)
+    >=> end
+
+let joinTeamMiddleware: M<Tuple3<User?, SubscriberState, User.TeamInviteCode>>
+  = writeStatus(.ok)
+    >=> end
+
+private func requireOwner<Z>(
```

Comment:
I needed to turn this into a function so that it could operate on `T3`s which allow me to reuse for the new team middlewares.